other-tests/character-class.js

```diff
@@ -0,0 +1,25 @@
+// Copyright 2017 Mathias Bynens. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Mathias Bynens
+description: >
+  Unicode property escapes must be supported in character classes.
+esid: sec-static-semantics-unicodematchproperty-p
+features: [regexp-unicode-property-escapes]
+---*/
+
+/[\p{Hex}]/u;
+assert(
+  /[\p{Hex}-\uFFFF]/u.test('-'),
+  'property escape in character class should not be parsed as the start of a range'
+);
+assert.throws(SyntaxError, () => /[\p{}]/u);
```

Comment:
isn't this an early error?

At least it is for `/\uGGGG/`, the test code won't be evaluated. For cases like this, you need to use the negative flags for each SyntaxError case. 

cc @anba @rwaldron. I might be wrong, you might be able to confirm what I'm saying here.